### DISCUSSION TITLE
Sending large files?

### DISCUSSION THREAD

**Discussion by BenWoodford** [sukona/Grapevine#233] [Q&A] [answered]
Is there any way to send large files with Grapevine (well, respond with them) without loading the _entire_ file into memory first? OutputStream seems to be what I'm looking for but there's not much documentation on it... do I need to add any headers, etc first for instance?

**Comment by scottoffen** [comment, marked as answer]
There are several overloads of `HttpResponse.SendResponse()` that will take a stream as a first parameter. Take a look at those, if you don't also pass in the ContentType and Encoding, you will have to set those on the response object yourself.

So, you could use `FileStream` or `MemoryStream` or which ever stream meets your specific needs.

Honestly, Grapevine 5 does a much better job of this.
